### ISSUE TITLE
Commands not stored

### ISSUE THREAD

**Issue by kobbejager** [jaedb/Iris#547]
**Describe the bug**
Commands disappear after restarting Mopidy

**To Reproduce**
1. Create a command and save it
2. Restart Mopidy (or the raspberry pi or pc)
3. Command(s) is/are gone

**Expected behavior**
Commands should be stored (server-side) and reloaded after boot

**Environment details:**
 - Raspbian Buster Lite
 - Firefox
 - Iris version 3.49.0

**Additional context**
Where on the server should these commands be stored anyway?

**Comment by kobbejager** [comment]
Mopidy's data_dir points to `$XDG_DATA_DIR/mopidy`, which translates to `/var/lib/mopidy` on my system. This directory exists and has owner mopidy:audio.

mopidy_iris/ext.conf sets the default data_dir for Iris on `$XDG_DATA_DIR/iris`. I have not overruled this in my mopidy.conf. This would point to `/var/lib/iris` if I'm not mistaken, which does not exist. If this directory has to be created by the user mopidy, this will evidently fail, because this user has no writing permissions on the /var/lib directory.

Could this be the reason?

**Comment by kobbejager** [comment]
Another follow-up: I have created /var/lib/iris with the appropriate rights, but it doesn't seem to solve the issue.

**Comment by jaedb** [comment]
What if you configure Iris to use a different directory entirely? It would look a little something like 
```
[iris]
enabled = true
data_dir = /homes/my-user/iris
```
Please note that how you're running Mopidy (in terminal vs as a service) will change what your actual config and paths resolve to. I'd suggest reading the [config docs](https://docs.mopidy.com/en/latest/config/#configuration-file-location) as it may help to better understand where this folder is expected to be on your environment.

**Comment by kingosticks** [comment]
I'm not sure if this exactly the problem but I've been meaning to ask about `iris/data_dir` and maybe I can clarify some things.

When using the Mopidy Debian package with the normal service configuration we provide, Mopidy's `core/data_dir` is explicitly set to var/lib/mopidy. You can see this when you run `sudo mopidyctl config`. (This non-default value is set in /usr/share/mopidy/conf.d/mopidy.conf but that doesn't matter to us here). The default value of $XDG_DATA_DIR/mopidy is for when running in the terminal. And I know you might also have your docker image with whatever value. 

But extensions don't need to worry about any of that. All extensions need to do is call `Extension.get_data_dir(config)` and they'll get a subdirectory within Mopidy's configured `core/data_dir` with the correct permissions ready to use. You don't need to have your own `iris/data_dir` config setting. If you do have your own special config setting then you must ensure user's set it correctly depending on how they run Mopidy, and you must also manage the creation (and deletion) of that directory yourself. 

Now, the version of the Iris files I have on my machine (presumably from the 2to3 migration help) uses `Extension.get_data_dir(config)` but I see now that this was [changed](https://github.com/jaedb/Iris/commit/37220ea523d6939751be050d910607a5292eb1fd) and the latest Iris code does not. It uses this `iris/data_dir` config value instead. Is this really what you need/want to do?

**Comment by jaedb** [comment]
Thanks heaps for the info - it certainly works different to how I had understood.
From memory I wanted a consistent folder path for Iris `data_dir`, predominantly for use in a Docker environment where users would almost certainly want this to be a persistent volume or bind mount. I think a best-of-both approach would be to default `data_dir` to null (or `""` if it must be a string) and use the intended `Extension.get_data_dir(config)`, but where a value is defined, we just use that (and assume users know what they're doing).
Does this sound like a sensible solution?

**Comment by kingosticks** [comment]
I don't follow why the Docker use-case requires this, won't the path within the container still be constant when using the normal Mopidy provided data_dir? But if you do need it then this works, just a bit more complicated.

**Comment by kobbejager** [comment]
For your information, I have set data_dir to /test (which has read/write permissions for all). When I create a command, this is now correctly stored in /test/commands.pkl and correctly reloaded when restarting mopidy. I guess I will be able to remove my data_dir override once Iris uses the correct path? Thanks.

@kingosticks I was overriding the default configuration, by adding the $XDG_... variables to the [core] section in /etc/mopidy/mopidy.conf (see below). Why? No particular reason, I guess I copied it from the [documentation](https://docs.mopidy.com/en/latest/config/#core-configuration). Shouldn't I have done this?
```
[core]
cache_dir = $XDG_CACHE_DIR/mopidy
config_dir = $XDG_CONFIG_DIR/mopidy
data_dir = $XDG_DATA_DIR/mopidy
```

**Comment by kingosticks** [comment]
@kobbejager, [we recommmend you only set config values where you explicitly need something other than the default](https://docs.mopidy.com/en/latest/config/#editing-the-configuration). And the default in this case depends on how how you run Mopidy so it's a bit more complicated. I doubt many people need to set anything in the `core` section.

**Comment by jaedb** [comment]
Released as at `3.50.0`
